Target (#4047)
webpack/webpack.js.org: src/content/configuration/target.md

`electron[[X].Y]-preload`  | Compile for [Electron](https://electronjs.org/) for renderer process, providing a target using `NodeTemplatePlugin` with `asyncChunkLoading` set to `true`, `FunctionModulePlugin` for browser environments and `NodeTargetPlugin` and `ExternalsPlugin` for CommonJS and Electron built-in modules.
`node[[X].Y]`              | Compile for usage in a Node.js-like environment (uses Node.js `require` to load chunks)
`node-webkit[[X].Y]`       | Compile for usage in WebKit and uses JSONP for chunk loading. Allows importing of built-in Node.js modules and [`nw.gui`](http://docs.nwjs.io/en/latest/) (experimental)
`nwjs[[X].Y]`              | The same as `node-webkit`
`web`                      | Compile for usage in a browser-like environment __(default)__
`webworker`                | Compile as WebWorker
`esX`                      | Compile for specified ECMAScript version. Examples: es5, es2020.

> `browserslist`             | Infer a platform and the ES-features from a browserslist-config

Reviewer: ```suggestion
`browserslist`             | Infer a platform and the ES-features from a browserslist-config __(default if browserlist config is available)__
```

> For example, when the `target` is set to `"electron-main"`, webpack includes multiple electron specific variables. For more information on which templates and externals are used, you can refer to webpack's [source code](https://github.com/webpack/webpack/blob/master/lib/WebpackOptionsApply.js#L78-L98).

Reviewer: ```suggestion
For example, when the `target` is set to `"electron-main"`, webpack includes multiple electron specific variables.
```

> A version of `node` or `electron` may be optionally specified.

Reviewer: ```suggestion
A version of `node` or `electron` may be optionally specified. This is denoted by the `[[X].Y]` in the table above.
```

__webpack.config.js__

```js
module.exports = {
  // ...
  target: 'node12.18'
};
```

It helps determinate ES-features that may be used to generate a runtime-code (all the chunks and modules are wrapped by runtime code).

#### `browserslist`

If a project has a browserslist config, then webpack will use it for:

- Determinate ES-features that may be used to generate a runtime-code.
- Infer an environment (e.g: `last 2 node versions` the same as `target: "node"` with some [`output.environment`](/configuration/output/#outputenvironment) settings).

Supported browserslist values:

> - `browserslist` - use automatically resolved browserslist config and environment (from the nearest `package.json` or `BROWSERLIST` environment variable)

Reviewer: ```suggestion
- `browserslist` - use automatically resolved browserslist config and environment (from the nearest `package.json` or `BROWSERSLIST` environment variable, see [browserslist documentation](https://github.com/browserslist/browserslist#queries) for details)
```

- `browserslist:modern` - use `modern` environment from automatically resolved browserslist config
- `browserslist:last 2 versions` - use an explicit browserslist query (config will be ignored)
- `browserslist:/path/to/config` - explicitly specify browserslist config
- `browserslist:/path/to/config:modern` - explicitly specify browserslist config and an environment

### `[string]`

When multiple targets are passed, then common subset of features will be used:

__webpack.config.js__

```js
module.exports = {
  // ...
  target: ['web', 'es5']
};
```

> Webpack will generate a runtime code for web platform and will use only ES5 features.

Reviewer 2: ```suggestion
webpack will generate a runtime code for web platform and will use only ES5 features.
```

Not all targets may be mixed for now.

__webpack.config.js__

```js
module.exports = {
  // ...
  target: ['web', 'node']
};
```

> Will cause an error. Webpack does not support universal target for now.

Reviewer 2: ```suggestion
Will cause an error. webpack does not support universal target for now.
```
